File: Бодлогууд/untitled/src/main/java/org/example/Spouse.java
package org.example;

import java.util.Date;

public class Spouse extends Person{
    private String  anniversary;

    Spouse(){
        this.anniversary  = "empty";
    }
    Spouse(String anniversary, String name, String ssnum, int age){

        this.anniversary = anniversary;
        this.name = name;
        this.ssnum = ssnum;
        this.age = age;
    }

  String getAnniversary(){
      return anniversary;
  }
  void setAnniversary(String anniversary){
      this.anniversary = anniversary;
  }
}

File: Бодлогууд/Лаб9/Spouse.cpp
#include "Spouse.h"

Spouse::Spouse() : Person()
{
    this->anniversaryDate = new char[strlen("unknown") + 1];
    strcpy(this->anniversaryDate, "unknown");
}
Spouse::  Spouse(char *anniversaryDate, char *name, char* ssnum, int age) : Person(name, ssnum, age)
{
    this->anniversaryDate = new char[strlen(anniversaryDate) + 1];
    strcpy(this->anniversaryDate, anniversaryDate);
}
char *Spouse::getAnniversaryDate()
{
    return this->anniversaryDate;
}
void Spouse::setAnniversaryDate(char *anniversaryDate)
{
    delete this->anniversaryDate;
    this->anniversaryDate = new char[strlen(anniversaryDate) + 1];
    strcpy(this->anniversaryDate, anniversaryDate);
}
Spouse::~Spouse(){
    delete anniversaryDate;
}